apps/frontend/src/utilities/delta_util.ts

```diff
@@ -163,6 +165,68 @@ export function get_eval_start_time(
   return null;
 }
 
+export function getResultsSetExecutionTime(
+  resultsSet: SourcedContextualizedEvaluation
+): number {
+  let time = 0;
+  resultsSet.contains.forEach((value) => {
+    value.contains.forEach((control) => {
+      control.hdf.segments?.forEach((segment) => {
+        time += segment.run_time || 0;
+      });
+    });
+  });
```

Comment:
There's already a function in the codebase to do the last layer of this check, (see `getControlRunTime` in ControlRowHeader). If you move that function to a more accessible location then this becomes 2 reduce statements wrapping getControlRunTime.